packages/easy-testing/src/DataProvider/StaticFixtureFinder.php

```diff
@@ -23,14 +31,25 @@ public static function yieldDirectory(string $directory, string $suffix = '*.php
     /**
      * @return SplFileInfo[]
      */
-    private static function findFilesInDirectory(string $directory, string $suffix): array
+    private static function findFilesInDirectory(string $directory, string $suffix, bool $isExclusive = false): array
     {
-        $finder = Finder::create()
-            ->in($directory)
-            ->files()
-            ->name($suffix);
-
+        $finder    = Finder::create()->in($directory)->files()->name($suffix);
         $fileInfos = iterator_to_array($finder);
+
+        if (! $isExclusive) {
+            $finderAll = Finder::create()->in($directory)->files();
+            foreach ($finderAll as $key => $fileInfoAll) {
+                $fileNameFromAll = $fileInfoAll->getFileName();
+                if (! isset($fileInfos[$key])) {
+                    throw new ShouldNotHappenException(sprintf(
+                        '"%s" has invalid suffix, use "%s" suffix instead',
+                        $fileNameFromAll,
+                        $suffix
+                    ));
+                }
+            }
```

Comment:
This should be a standalone method, so it's easier to maintain in the future

Reply:
implemented.